src/main/java/spoon/reflect/visitor/ImportScannerImpl.java

```diff
@@ -218,7 +226,44 @@ private boolean addImport(CtTypeReference<?> ref) {
 		if (imports.containsKey(ref.getSimpleName())) {
 			return isImported(ref);
 		}
+		// don't import unnamed package elements
+		if (ref.getPackage() == null || ref.getPackage().isUnnamedPackage()) {
+			return false;
+		}
+		if (!ref.getPackage().getSimpleName().equals("java.lang")) {
+			if (classNamePresentInJavaLang(ref)) {
```

Comment:
those two consecutive ifs look like an oxymoron. can we rewrite them in a more maintainable way?